pkg/destroy/gcp/network.go

```diff
@@ -0,0 +1,647 @@
+			return err
+		}
+	}
+	return nil
+}
+
+// getInternalLBInstanceGroups finds instance groups created for kube cloud controller
+// internal load balancers. They should be named "k8s-ig--{clusterid}":
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_naming.go#L33-L40
+// where clusterid is an 8-char id generated and saved in a configmap named
+// "ingress-uid" in kube-system, key "uid":
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_clusterid.go#L210-L238
+// TODO: Have the installer generate this UID, save it to metadata and populate it in the cluster. With that, we can just directly get
+// an instanceGroup named "k8s-ig--{clusterid}"
+// For now, we look into each instance group and determine if it contains instances prefixed with the cluster's infra ID
+func (o *ClusterUninstaller) getInternalLBInstanceGroups() ([]nameAndZone, error) {
+	candidates, err := o.listInstanceGroups("name eq \"k8s-ig--.*\"")
+	if err != nil {
+		return nil, err
+	}
+	if len(candidates) == 0 {
+		return nil, nil
+	}
+
+	igName := ""
+	for _, ig := range candidates {
+		instances, err := o.listInstanceGroupInstances(ig)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get internal LB instance group instances for %s", ig.name)
+		}
+		if o.areAllClusterInstances(instances) {
+			igName = ig.name
+			break
+		}
+	}
+	igs := []nameAndZone{}
+	if len(igName) > 0 {
+		for _, ig := range candidates {
+			if ig.name == igName {
+				igs = append(igs, ig)
+			}
+		}
+	}
+
+	return igs, nil
+}
+
+func (o *ClusterUninstaller) areAllClusterInstances(instances []nameAndZone) bool {
+	for _, instance := range instances {
+		if !o.isClusterResource(instance.name) {
+			return false
+		}
+	}
+	return true
+}
+
+func (o *ClusterUninstaller) getInstanceGroupURL(ig nameAndZone) string {
+	return fmt.Sprintf("%s%s/zones/%s/instanceGroups/%s", o.computeSvc.BasePath, o.ProjectID, ig.zone, ig.name)
+}
+
+func (o *ClusterUninstaller) listBackendServicesForInstanceGroups(igs []nameAndZone) ([]string, error) {
+	urls := sets.NewString()
+	for _, ig := range igs {
+		urls.Insert(o.getInstanceGroupURL(ig))
+	}
+	return o.listBackendServices("items(name,backends)", "", func(item *compute.BackendService) bool {
+		if len(item.Backends) == 0 {
+			return false
+		}
+		for _, backend := range item.Backends {
+			if !urls.Has(backend.Group) {
+				return false
+			}
+		}
+		return true
+	})
+}
+
+// deleteInternalLoadBalancer follows a similar cleanup procedure as:
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_internal.go#L222
+// TODO: add cleanup for shared mode resources (determine if it's supported in 4.2)
+func (o *ClusterUninstaller) deleteInternalLoadBalancer(clusterID string, loadBalancerName string) error {
```

Comment:
would it be useful to convert these resources to something that our cleanup code for these resource type picks up rather than doing a synchronized delete here...?

Reply:
not sure how, most of these cannot be labeled or renamed

Reply:
hmm.. that's sad. looks like this wil do for now.